src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5517,4 +5518,14 @@ public void checkLink() throws SecurityException {
             security.checkLink(dllName);
         }
     }
+    
+    /**
+     * Although {@link ActivityCorrelator} used in different part of programs releasing connection might be good point to release {@link ThreadLocal}
+     * resource. We are releasing {@link ThreadLocal} resource in {@link SQLServerConnection#close()} too.
+     * @since 6.1.8
+     */
+    @Override
+    protected void finalize() throws Throwable {
```

Comment:
[Oracle recommends against using finalize](https://stuartmarks.wordpress.com/2017/04/17/deprecation-of-object-finalize/) to the point where it's deprecated in Java 9. They recommend try-with-resources or `PhantomReference` instead.